src/couch/src/couch_password_server.erl

```diff
@@ -0,0 +1,80 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_password_server).
+
+-behaviour(gen_server).
+
+-include_lib("couch/include/couch_db.hrl").
+
+-export([start_link/0]).
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    code_change/3
+]).
+
+-export([hash/2]).
+
+-record(state, {}).
+
+%%%===================================================================
+%%% Public functions
+%%%===================================================================
+
+-spec hash(IsAsyncCall :: boolean(), Persist :: boolean()) -> Reply :: term().
+hash(IsAsyncCall, Persist) ->
+    case IsAsyncCall of
+        true ->
+            gen_server:cast(?MODULE, {hash_passwords, Persist});
+        false ->
+            gen_server:call(?MODULE, {hash_passwords, Persist})
+    end.
+
+%%%===================================================================
+%%% Spawning and gen_server implementation
+%%%===================================================================
+
+start_link() ->
+    couch_log:info("Password Hasher Process is starting.~n", []),
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init(_Args) ->
+    {ok, #state{}}.
+
+handle_call({hash_passwords, Persist}, _From, _State) ->
+    {reply, hash_admin_passwords(Persist), _State};
+handle_call(Msg, _From, State = #state{}) ->
+    {stop, {invalid_call, Msg}, {invalid_call, Msg}, State}.
+
+handle_cast({hash_passwords, Persist}, _State) ->
```

Comment:
`_State` is used, it should not have an underscore. As above, let's use `#state{} = State`